osu.Server.Queues.ScorePump/RecalculatePlaylistTotalScoresCommand.cs

```diff
@@ -0,0 +1,179 @@
+
+        public async Task<int> OnExecuteAsync(CancellationToken cancellationToken)
+        {
+            using (var db = Queue.GetDatabaseConnection())
+            {
+                var playlistItems = await db.QueryAsync<MultiplayerPlaylistItem>("SELECT * FROM multiplayer_playlist_items WHERE room_id = @PlaylistId", new
+                {
+                    PlaylistId
+                });
+
+                foreach (var item in playlistItems)
+                {
+                    MultiplayerScore[] scores = (await db.QueryAsync<MultiplayerScore>($"SELECT * FROM multiplayer_scores WHERE playlist_item_id = {item.id}")).ToArray();
+
+                    foreach (var score in scores)
+                    {
+                        long? scoreBefore = score.total_score;
+
+                        if (scoreBefore == null || !score.passed)
+                            continue;
+
+                        Console.WriteLine($"Reprocessing score {score.id} from playlist {item.id}");
+
+                        long scoreAfter = ensureCorrectTotalScore(score, item);
+
+                        if (scoreAfter != scoreBefore)
+                        {
+                            Console.WriteLine($"Score requires update ({scoreBefore} -> {scoreAfter})");
+                            score.total_score = scoreAfter;
+
+                            // TODO: destroy production
+                            // await db.UpdateAsync(score);
+                        }
+                        else
+                            Console.WriteLine("Score is correct");
+                    }
+                }
+            }
+
+            Console.WriteLine("Finished.");
+            return 0;
+        }
+
+        private long ensureCorrectTotalScore(MultiplayerScore score, MultiplayerPlaylistItem playlistItem)
+        {
+            Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(playlistItem.ruleset_id);
+
+            HitResult maxRulesetJudgement = ruleset.GetHitResults().First().result;
+
+            Dictionary<HitResult, int> statistics = JsonConvert.DeserializeObject<Dictionary<HitResult, int>>(score.statistics)!;
+
+            List<HitResult> allHits = statistics
+                                      .SelectMany(kvp => Enumerable.Repeat(kvp.Key, kvp.Value))
+                                      .ToList();
+
+            var maximumStatistics = new Dictionary<HitResult, int>();
+
+            foreach (var groupedStats in allHits
+                                         .Select(r => getMaxJudgementFor(r, maxRulesetJudgement))
+                                         .GroupBy(r => r))
+            {
+                maximumStatistics[groupedStats.Key] = groupedStats.Count();
+            }
+
+            APIMod[] roomMods = JsonConvert.DeserializeObject<APIMod[]>(playlistItem.required_mods)!;
+            APIMod[] scoreMods = JsonConvert.DeserializeObject<APIMod[]>(score.mods)!;
+
+            foreach (var m in roomMods)
```

Comment:
This is just a safety I added to be sure that the scores include any room mods, since it was easier to add this than check it's the case.